KMMYCharts/src/commonMain/kotlin/co/yml/kmm/charts/ui/piechart/charts/DonutPieChart.kt

```diff
@@ -0,0 +1,208 @@
+import co.yml.kmm.charts.ui.piechart.utils.convertTouchEventPointToAngle
+import co.yml.kmm.charts.ui.piechart.utils.proportion
+import co.yml.kmm.charts.ui.piechart.utils.sweepAngles
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
+
+
+/**
+ * Compose function for Drawing Donut chart
+ * @param modifier : All modifier related property
+ * @param pieChartData: data list for the pie chart
+ * @param pieChartConfig: configuration for the pie chart
+ * @param onSliceClick(pieChartData.Slice)->Unit: The event that captures the click
+ */
+@OptIn(ExperimentalTextApi::class)
+@ExperimentalMaterialApi
+@Composable
+internal fun DonutPieChart(
+    modifier: Modifier,
+    pieChartData: PieChartData,
+    pieChartConfig: PieChartConfig,
+    onSliceClick: (PieChartData.Slice) -> Unit = {}
+) {
+    // Sum of all the values
+    val sumOfValues = pieChartData.totalLength
+
+    // Calculate each proportion value
+    val proportions = pieChartData.slices.proportion(sumOfValues)
+
+    // Convert each proportions to angle
+    val sweepAngles = proportions.sweepAngles()
+
+    val progressSize = mutableListOf<Float>()
+    progressSize.add(sweepAngles.first())
+
+    for (x in 1 until sweepAngles.size) {
+        progressSize.add(sweepAngles[x] + progressSize[x - 1])
+    }
+
+    var activePie by rememberSaveable {
+        mutableStateOf(-1)
+    }
+    val textMeasurer = rememberTextMeasurer()
+    val accessibilitySheetState =
+        rememberModalBottomSheetState(initialValue = ModalBottomSheetValue.Hidden)
+    val scope = rememberCoroutineScope()
+    val isTalkBackEnabled = false
+    if (accessibilitySheetState.isVisible && isTalkBackEnabled
+        && pieChartConfig.accessibilityConfig.shouldHandleBackWhenTalkBackPopUpShown
+    ) {
+        // todo handle accessibility bottomsheet dismissal
+    }
+    Surface(
+        modifier = modifier
+    ) {
+        BoxWithConstraints(
+            modifier = modifier
+                .aspectRatio(1f)
+                .semantics {
+                    contentDescription = pieChartConfig.accessibilityConfig.chartDescription
+                }
+                .clickable {
+                    if (isTalkBackEnabled) {
+                        scope.launch {
+                            accessibilitySheetState.animateTo(
+                                ModalBottomSheetValue.Expanded
+                            )
+                        }
+                    }
+                }) {
+
+            val sideSize = if (constraints.maxWidth < constraints.maxHeight) constraints.maxWidth else constraints.maxHeight
+            val padding = (sideSize * pieChartConfig.chartPadding) / 100f
+            val size = Size(sideSize.toFloat() - padding, sideSize.toFloat() - padding)
+
+            val pathPortion = remember {
+                Animatable(initialValue = 0f)
+            }
+
+            if (pieChartConfig.isAnimationEnable) {
+                LaunchedEffect(key1 = Unit) {
```

Comment:
@harshakiran3 Here instead of Unit can we use compose lifecycle object?

Reply:
It is not supported in KMM compose.